### PR TITLE
fix: External avatar url provider for rooms not respected.

### LINKED ISSUE #28351: Room Avatar External Provider URL is not working
Hi I have a local deployment of rocketchat version 5.4.1 and I want to use `Room Avatar External Provider URL` feature to load all the room avatars from my local asset server which is running on port 3001. When I set the url to `http://localhost:3001/api/chat-room-avatar/{roomId}` it does not request to my server. `Avatar External Provider URL` is working fine but other one is not working. can you help me?

### PR DESCRIPTION
**Pull request by gabriellsh**
## Issue(s)

Closes #28351

## Steps to test or reproduce

- Get a link to an image
- go to `Admin -> Settings -> Accounts -> Avatar -> Room Avatar External Provider URL`
- paste the link there
- save

All rooms should have this link as avatar.
## Further comments

[TC-635]

[TC-635]: https://rocketchat.atlassian.net/browse/TC-635?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/app/utils/lib/getRoomAvatarURL.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { settings } from '../../settings';
+
+export const getRoomAvatarURL = ({ roomId, cache = '' }: { roomId: IRoom['_id']; cache: IRoom['avatarETag'] }) => {
+	const externalSource = (settings.get('Accounts_RoomAvatarExternalProviderUrl') || '').trim().replace(/\/$/, '');
+	if (externalSource !== '') {
+		return externalSource.replace('{roomId}', roomId);
+	}
+	if (roomId == null) {
```
Comment (tassoevan) [inline apps/meteor/app/utils/lib/getRoomAvatarURL.ts:15]: This conditional is too old-fashioned and not type-safe.

File: apps/meteor/client/lib/rooms/roomTypes/private.ts [hunk 2]
```diff
@@ -87,7 +87,7 @@ roomCoordinator.add(
 		},
 
 		getAvatarPath(room) {
-			return getAvatarURL({ roomId: room._id, cache: room.avatarETag }) as string;
+			return getRoomAvatarURL({ roomId: room._id, cache: room.avatarETag }) as string;
```
Comment (tassoevan) [inline apps/meteor/client/lib/rooms/roomTypes/private.ts:90]: Why do it need to assert to `string`?

File: apps/meteor/client/lib/rooms/roomTypes/public.ts [hunk 3]
```diff
@@ -87,7 +87,7 @@ roomCoordinator.add(
 		},
 
 		getAvatarPath(room) {
-			return getAvatarURL({ roomId: room._id, cache: room.avatarETag }) as string;
+			return getRoomAvatarURL({ roomId: room._id, cache: room.avatarETag }) as string;
```
Comment (tassoevan) [inline apps/meteor/client/lib/rooms/roomTypes/public.ts:90]: Why do it need to assert to `string`?

**Review by tassoevan — CHANGES_REQUESTED** [review]

**Review by tassoevan — APPROVED** [review]

**Comment by amirAlamian** [comment]
did this feature merge into master branch?

**Pull request merged**